app/services/katello/registration_manager.rb

```diff
@@ -79,6 +79,9 @@ def validate_hosts(hosts, organization, host_name, host_uuid, host_uuid_overridd
           if host.name == host_name
             unless Setting[:host_profile_assume] || host.subscription_facet.nil? || host.build || host_uuid_overridden
               current_dmi_uuid = host.subscription_facet.dmi_uuid
+              if !Setting[:host_re_register] && !host.build && current_dmi_uuid
```

Comment:
I see the goal in this change but I think it needs to be handled in a different way for two reasons:

- host_re_register (as you've pointed out) is already the current behavior, so adding a setting for it feels strange
- the fact that the setting is bypassed if the host is in build mode seems to be a bit of an undocumented loophole


potential solution:

Rename the setting to `host_re_register_build_only` with the text being something like "Only allow previously registered hosts to re-register if they are in build mode"

Then the logic becomes something like:

```
if !host.build && Setting[:host_re_register_build_only] #putting the build check first avoids DB hit on setting lookup if host is in build mode already
 # raise error
end

#dmi uuid uniqueness enforcement happens down here
```

Reply:
This makes **_a lot_** more sense!

I'll see about getting this written up as listed here once #9312 is merged.

Reply:
I've taken a stab at the requested changes